native/src/main/java/io/ballerina/stdlib/azure/functions/FunctionCallback.java

```diff
@@ -81,6 +82,50 @@ public void notifySuccess(Object result) {
 //        }
         if (this.annotations.get(0).equals("QueueOutput") || this.annotations.get(0).equals("CosmosDBOutput")) {
             mapValue.put(StringUtils.fromString("outMsg"), result);
+        } else if (this.annotations.get(0).equals("HTTPOutput")) {
+            if (isHTTPResponse(result)) {
+                BObject status = (BObject) (((BMap) result).get(StringUtils.fromString("status")));
+                Object statusCode = Long.toString(status.getIntValue(StringUtils.fromString("code")));
+                statusCode = StringUtils.fromString((String) statusCode);
+
+                BMap<BString, Object> bodyMap =
+                        ValueCreator.createMapValue(TypeCreator.createMapType(PredefinedTypes.TYPE_ANYDATA));
+                bodyMap.put(StringUtils.fromString("statusCode"), statusCode);
+
+                if (((BMap) result).containsKey(StringUtils.fromString("body"))) {
+                    Object body = ((BMap) result).get(StringUtils.fromString("body"));
+                    bodyMap.put(StringUtils.fromString("body"), body);
+                }
+
+                if (((BMap) result).containsKey(StringUtils.fromString("headers"))) {
```

Comment:
shall we extract this(and other) complex logic into a function to improve the readability?